src/openrct2-ui/windows/RideConstruction.cpp

```diff
@@ -2540,8 +2541,9 @@ void window_ride_construction_update_enabled_track_pieces()
     if (rideEntry == nullptr)
         return;
 
-    int32_t rideType = (_currentTrackAlternative & RIDE_TYPE_ALTERNATIVE_TRACK_TYPE) ? RideData4[ride->type].alternate_type
-                                                                                     : ride->type;
+    int32_t rideType = (_currentTrackAlternative & RIDE_TYPE_ALTERNATIVE_TRACK_TYPE)
+        ? RideTypeDescriptors[ride->type].AlternateType
+        : ride->type;
```

Comment:
Duplication of the `ride_get_alternative_type()` function.

Reply:
Is this a note for future or do you intend to fix this in the patch?

Reply:
I intend to fix it in this patch.